core/tests/puppeteer-acceptance-tests/user-utilities/question-admin-utils.js

```diff
@@ -0,0 +1,317 @@
+    const displayedUsername = await this.page.$eval(
+      viewContributorReviewQuestionsResult,
+      element => element.innerText);
+    if (!displayedUsername.includes('Not-allowed')) {
+      throw new Error(
+        `${username} has rights for reviewing questions!`);
+    } else {
+      showMessage(
+        `${username} doesn't have rights for reviewing questions.`);
+    }
+  }
+
+  /**
+   * Function to display contribution rights by user
+   * and check if the user doesn't have the right to submit questions
+   * @param {string} username - the username of the user to view.
+   */
+  async verifyUserCannotSubmitQuestions(username) {
+    await this.select(viewContributorFilterMethodSelect, usernameMethodValue);
+    await this.type(viewContributerUsernameInput, username);
+    await this.clickOn(viewContributorSubmitButton);
+
+    await this.page.waitForNetworkIdle();
+
+    await this.page.waitForSelector(viewContributorSubmitQuestionResult);
+    const displayedUsername = await this.page.$eval(
+      viewContributorSubmitQuestionResult,
+      element => element.innerText);
+    if (!displayedUsername.includes('Not-allowed')) {
+      throw new Error(
+        `${username} has rights for submitting questions!`);
+    } else {
+      showMessage(
+        `${username} doesn't have rights for submitting questions.`);
+    }
+  }
+
+  /**
+   * Function to display reviewing questions rights by role
+   * and check if the user is displayed as a question reviewer
+   * @param {string} username - the user expected to be displayed.
+   */
+  async verifyQuestionReviewersIncludeUser(username) {
+    await this.select(viewContributorFilterMethodSelect, roleMethodValue);
+    await this.select(viewContributorCategorySelect, reviewQuestionRightValue);
+    await this.clickOn(viewContributorSubmitButton);
+
+    await this.page.waitForNetworkIdle();
+
+    await this.page.waitForSelector(viewRoleUserResult);
+    const displayedUsers = await this.page.$eval(
+      viewRoleUserResult,
+      element => element.innerText
+    );
+    if (!displayedUsers.includes(username)) {
+      throw new Error(
+        `${username} does not have rights for reviewing questions!`);
+    }
+  }
+
+  /**
+   * Function to display reviewing questions rights by role
+   * and check if the user is displayed as a question submitter
+   * @param {string} username - the user expected to be displayed.
+   */
+  async verifyQuestionSubmittersIncludeUser(username) {
+    await this.select(viewContributorFilterMethodSelect, roleMethodValue);
+    await this.select(viewContributorCategorySelect, submitQuestionRightValue);
+    await this.clickOn(viewContributorSubmitButton);
+
+    await this.page.waitForNetworkIdle();
+
+    await this.page.waitForSelector(viewRoleUserResult);
+    const displayedUsers = await this.page.$eval(
+      viewRoleUserResult,
+      element => element.innerText
+    );
```

Comment:
I think it would be a good idea to have an async private function that returns the list of displayed users for the submit case, and another that does the same for the review case -- then these public functions can use the results from those private ones.  That way you avoid duplicating large chunks of code, and the returned list of users seems like a sensible common point to factor the logic out.

Reply:
Usually for functions that return something I would start them with `get`. So perhaps getDisplayedListOfQuestionSubmitters() or similar -- thoughts?

That way, it's clear it's actually returning something and not just a passive "viewing" action.

Reply:
Done

Reply:
I created two functions that return the list of displayed users with rights of question reviewing and submitting, and also a new function called contributionStatusForUser that takes the username as an argument and returns the status of the contribution rights of the user.